src/TSTransformer/nodes/statements/transformForOfStatement.ts

```diff
@@ -447,6 +493,14 @@ export function transformForOfStatement(state: TransformState, node: ts.ForOfSta
 		}
 	}
 
+	if (ts.isCallExpression(node.expression)) {
+		const expType = state.typeChecker.getNonOptionalType(state.getType(node.expression.expression));
```

Comment:
Would we ever allow `$range | undefined` as a type here? I don't see why the `getNonOptionalType` is necessary.